core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/saved-workflow-section.component.spec.ts

```diff
@@ -174,15 +193,117 @@ describe("SavedWorkflowSectionComponent", () => {
     component.dashboardWorkflowEntries = [];
     component.dashboardWorkflowEntries = component.dashboardWorkflowEntries.concat(testWorkflowEntries);
     component.dateSort();
-    const SortedCase = component.dashboardWorkflowEntries.map(item => item.workflow.creationTime);
-    expect(SortedCase).toEqual([1, 3, 3, 3, 4]);
+    const SortedCase = component.dashboardWorkflowEntries.map(item => item.workflow.name);
+    expect(SortedCase).toEqual(["workflow 1", "workflow 3", "workflow 2", "workflow 5", "workflow 4"]);
   });
 
   it("lastEditSortTest", () => {
     component.dashboardWorkflowEntries = [];
     component.dashboardWorkflowEntries = component.dashboardWorkflowEntries.concat(testWorkflowEntries);
     component.lastSort();
-    const SortedCase = component.dashboardWorkflowEntries.map(item => item.workflow.lastModifiedTime);
-    expect(SortedCase).toEqual([2, 3, 4, 6, 8]);
+    const SortedCase = component.dashboardWorkflowEntries.map(item => item.workflow.name);
+    expect(SortedCase).toEqual(["workflow 1", "workflow 2", "workflow 3", "workflow 4", "workflow 5"]);
   });
+
+  //OLD TEST CASES for searching based on input bar string
```

Comment:
please add the corresponding test cases after we change from search on string to search on dropdown menus. test cases are very important

There are two ways you can write the test cases:
1. first way is to emulate a dropdown selectin on the element, you can use `fixture.debugElement`, this is a bit harder, check https://angular.io/guide/testing-components-basics#debugelement
2. second way is to just change the corresponding models, for example, just do `owners.append(...)`, then manually call search, then verify the search result.

first way is a bit harder. I suggest you to try both ways.